inc/saiswitch.h

```diff
@@ -1995,6 +2260,62 @@ typedef void (*sai_switch_state_change_notification_fn)(
         _In_ sai_object_id_t switch_id,
         _In_ sai_switch_oper_status_t switch_oper_status);
 
+/**
+ * @brief Platform specific device register read access
+ *
+ * This API provides platform adaption in 2 ways
+ *
+ * 1# Provides platform adaption functionality to access device
+ * registers from driver. This is mandatory to pass as attribute to
+ * sai_create_switch when driver implementation does not support register access
+ * by device file system directly.
+ * 2# Provides device access API from ASIC SDK.
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_SWITCH
+ *
+ * @param[in] platform_context Platform context information.
+ * @param[in] switch_id Switch Id
+ * @param[in] device_addr Device address(PHY/lane/port MDIO address)
+ * @param[in] start_reg_addr Starting register address to read
+ * @param[in] number_of_registers Number of consecutive registers to read
+ * @param[out] reg_val Register read values
+ */
+typedef sai_status_t (*sai_switch_register_read_fn)(
+        _In_ uint64_t platform_context,
+        _In_ sai_object_id_t switch_id,
```

Comment:
what is this switch_id?

Reply:
@kamil , @lguohan -  Yes, we can remove switch_id, Because while switch_create. we do not have switch_id created in NOS.

Reply:
i was thinking more like this, if same callback function would be used in many gearboxes, then switch_id would uniquely identify which switch we want to talk to, but this role should also be the same for platform_context, so one of them seems to be redundant

Reply:
just to clarify after meetings, switch id will be removed from this callback since that should be passed to other SAI as a callback, and it would make no sens for other npu understanding switch id from other npu